src/pf-popover/pf-popover.component.js

```diff
@@ -0,0 +1,421 @@
+   */
+  get containerSelector() {
+    return this._containerSelector;
+  }
+
+  /**
+   * Set the popover containerSelector
+   *
+   * @param {string} value The container element selector
+   */
+  set containerSelector(value) {
+    if (this._containerSelector !== value) {
+      this._containerSelector = value;
+      this._container = document.querySelector(this._containerSelector);
+      this.setAttribute('containerSelector', value);
+    }
+  }
+
+  /**
+   * Get the animation duration
+   *
+   * @returns {string} The animation duration
+   */
+  get duration() {
+    return this._duration;
+  }
+
+  /**
+   * Set the animation duration
+   *
+   * @param {string} value The animation duration
+   */
+  set duration(value) {
+    if (this._duration !== value) {
+      this._duration = value;
+      this.setAttribute('duration', value);
+    }
+  }
+
+  /**
+   * Get the animation delay
+   *
+   * @returns {string} The animation delay
+   */
+  get delay() {
+    return this._duration;
+  }
+
+  /**
+   * Set the animation delay
+   *
+   * @param {string} value The animation delay
+   */
+  set delay(value) {
+    if (this._delay !== value) {
+      this._delay = value;
+      this.setAttribute('delay', value);
+    }
+  }
+
+  /**
+   * Get the placement this._placement
+   *
+   * @returns {string} The placement this._placement left, top, bottom, right
+   */
+  get placement() {
+    return this._placement;
+  }
+
+  /**
+   * Set placement this._placement
+   *
+   * @param {string} value The placement this._placement left, top, bottom, right
+   */
+  set placement(value) {
+    if (this._placement !== value) {
+      this._placement = value;
+      this.setAttribute('placement', value);
+    }
+  }
+
+  /**
+  * Get the targetSelector
+  *
+  * @returns {string} The target element selector
+  */
+  get targetSelector() {
+    return this._targetSelector;
+  }
+
+  /**
+   * Set targetSelector
+   *
+   * @param {string} value The target element selector
+   */
+  set targetSelector(value) {
+    if (this._targetSelector !== value) {
+      this._targetSelector = value;
+      this._target = document.querySelector(this._targetSelector);
+      this.setAttribute('targetSelector', value);
+    }
+  }
+
+  /**
+   * Get the popoverTitle
+   *
+   * @return {string} the title of popover
+   */
+  get popoverTitle() {
+    return this._popoverTitle;
+  }
+
+  /**
+   * Set popoverTitle
+   *
+   * @param {string} value The title of popover
+   */
+  set popoverTitle(value) {
+    if (this._popoverTitle !== value) {
+      this._popoverTitle = value;
+      this.setAttribute('popoverTitle', value);
+    }
+  }
+
+  /**
+     * The popover open method
+     */
+  open() {
+    clearTimeout(this._timer);
+    this._timer = setTimeout(() => {
+      if (this.popover === null) {
+        this._createPopover();
+        this._stylePopover();
+        this._checkPlacement();
+        this._showPopover();
```

Comment:
Hey @sahil143, I noticed the same thing after discussing this w/ @cdcabrera. It appears that the `popover on left but not in view` example showcases this. It seems to be creating a `popover left fade` div unnecessarily (when the popover is actually on the right and inside another div). Let's try to correct the logic creating this `left` div in this scenario...

![screen shot 2017-07-18 at 9 58 36 am](https://user-images.githubusercontent.com/4237045/28321044-2f98eec2-6ba0-11e7-848e-27551aa08355.png)